### ISSUE TITLE
React Native throws error upgrading from `0.2.25` to `0.2.30`

### ISSUE THREAD

**Issue by crutchcorn** [typeorm/typeorm#7330]
While upgrading my React Native app's dependencies:

https://github.com/oceanbit-dev/GitShark/tree/7c30a38fa977191ef2e1c8507e639adae1eb054d

And updating TypeORM from `0.2.25` to `0.2.30`, I get an error when trying to open a repo after cloning. These associates with the following commands in TypeORM:

```
[Sun Jan 31 2021 04:07:01.680]  INFO     query:  DROP TABLE "repo"
[Sun Jan 31 2021 04:07:01.800]  INFO     query:  ALTER TABLE "temporary_repo" RENAME TO "repo"
[Sun Jan 31 2021 04:07:01.107]  INFO     query:  COMMIT
[Sun Jan 31 2021 04:07:01.348]  INFO     query:  SELECT "Repo"."id" AS "Repo_id", "Repo"."name" AS "Repo_name", "Repo"."currentbranchname" AS "Repo_currentbranchname", "Repo"."commitstopull" AS "Repo_commitstopull", "Repo"."commitstopush" AS "Repo_commitstopush", "Repo"."path" AS "Repo_path", "Repo"."lastupdated" AS "Repo_lastupdated" FROM "repo" "Repo"
```

It seems the error stems from this regex creation:

https://github.com/typeorm/typeorm/blob/master/src/query-builder/QueryBuilder.ts#L626-L634

Modifying line to help with debugging:

```javascript
const fullRegex = ("(?<=[ =(]|^.{0})" +
(escapeRegExp(replaceAliasNamePrefix) + "(" + replacementKeys.map(escapeRegExp).join("|") + ")") +
"(?=[ =),]|.{0}$)", "gm");

console.log({statement, replaceAliasNamePrefix, replacementKeys, fullRegex} );

statement = statement.replace(new RegExp("(?<=[ =(]|^.{0})" +
    (escapeRegExp(replaceAliasNamePrefix) + "(" + replacementKeys.map(escapeRegExp).join("|") + ")") +
    "(?=[ =),]|.{0}$)", "gm"), function (_, p) {
    return "" + replacementAliasNamePrefix + _this.escape(replacements[p]);
});
```

I get the following stacktrace with the error in question:

<details>
<summary> `Invalid regular expression: invalid group specifier name` </summary>

[Sun Jan 31 2021 04:07:01.680]  INFO     query:  DROP TABLE "repo"
[Sun Jan 31 2021 04:07:01.800]  INFO     query:  ALTER TABLE "temporary_repo" RENAME TO "repo"
[Sun Jan 31 2021 04:07:01.107]  INFO     query:  COMMIT
[Sun Jan 31 2021 04:07:01.348]  INFO     query:  SELECT "Repo"."id" AS "Repo_id", "Repo"."name" AS "Repo_name", "Repo"."currentbranchname" AS "Repo_currentbranchname", "Repo"."commitstopull" AS "Repo_commitstopull", "Repo"."commitstopush" AS "Repo_commitstopush", "Repo"."path" AS "Repo_path", "Repo"."lastupdated" AS "Repo_lastupdated" FROM "repo" "Repo"
[Sun Jan 31 2021 04:07:02.469]  LOG      {"fullRegex": "gm", "replaceAliasNamePrefix": "Repo.", "replacementKeys": ["commits", "commits.id", "commits.oid", "id", "name", "currentbranchname", "commitstopull", "commitstopush", "path", "lastupdated", "currentBranchName", "commitsToPull", "commitsToPush", "lastUpdated"], "statement": "Repo_Repo__commits.repoId=Repo.id"}
[Sun Jan 31 2021 04:07:02.513]  LOG      {"error": {"message": "Invalid regular expression: invalid group specifier name", "name": "SyntaxError", "stack": "RegExp@[native code]
_loop_1@http://localhost:8081/index.bundle?platform=android&dev=true&minify=false:203697:51
replacePropertyNames@http://localhost:8081/index.bundle?platform=android&dev=true&minify=false:203709:18
http://localhost:8081/index.bundle?platform=android&dev=true&minify=false:201490:169
map@[native code]
createJoinExpression@http://localhost:8081/index.bundle?platform=android&dev=true&minify=false:201446:56
getQuery@http://localhost:8081/index.bundle?platform=android&dev=true&minify=false:200395:39
getQueryAndParameters@http://localhost:8081/index.bundle?platform=android&dev=true&minify=false:203432:32
http://localhost:8081/index.bundle?platform=android&dev=true&minify=false:201923:65
step@http://localhost:8081/index.bundle?platform=android&dev=true&minify=false:165591:27
http://localhost:8081/index.bundle?platform=android&dev=true&minify=false:165521:22
http://localhost:8081/index.bundle?platform=android&dev=true&minify=false:165493:75
tryCallTwo@http://localhost:8081/index.bundle?platform=android&dev=true&minify=false:26964:9
doResolve@http://localhost:8081/index.bundle?platform=android&dev=true&minify=false:27128:25
Promise@http://localhost:8081/index.bundle?platform=android&dev=true&minify=false:26987:14
__awaiter@http://localhost:8081/index.bundle?platform=android&dev=true&minify=false:165472:38
loadRawResults@http://localhost:8081/index.bundle?platform=android&dev=true&minify=false:201917:35
http://localhost:8081/index.bundle?platform=android&dev=true&minify=false:201832:45
step@http://localhost:8081/index.bundle?platform=android&dev=true&minify=false:165591:27
http://localhost:8081/index.bundle?platform=android&dev=true&minify=false:165521:22
http://localhost:8081/index.bundle?platform=android&dev=true&minify=false:165493:75
tryCallTwo@http://localhost:8081/index.bundle?platform=android&dev=true&minify=false:26964:9
doResolve@http://localhost:8081/index.bundle?platform=android&dev=true&minify=false:27128:25

</details>

### My Environment

| Dependency          | Version  |
| ---                 | ---      |
| Operating System    |          |
| ReactNative version     | v0.63.4 | 
| Typescript version  | v3.9.7 | 
| TypeORM version     | v0.2.30 | 

### Relevant Database Driver(s)

- [ ] `aurora-data-api`
- [ ] `aurora-data-api-pg`
- [ ] `better-sqlite3`
- [ ] `cockroachdb`
- [ ] `cordova`
- [ ] `expo`
- [ ] `mongodb`
- [ ] `mysql`
- [ ] `nativescript`
- [ ] `oracle`
- [ ] `postgres`
- [x] `react-native`
- [ ] `sap`
- [ ] `sqlite`
- [ ] `sqlite-abstract`
- [ ] `sqljs`
- [ ] `sqlserver`

### Are you willing to resolve this issue by submitting a Pull Request?

- [ ] Yes, I have the time, and I know how to start.
- [x] Yes, I have the time, but I don't know how to start. I would need guidance.
- [ ] No, I don't have the time, although I believe I could do it if I had the time...
- [ ] No, I don't have the time and I wouldn't even know how to start.

**Comment by stretchkennedy** [comment]
Looks like a duplicate of https://github.com/typeorm/typeorm/pull/7270, which was fixed in `0.2.31`.

**Comment by crutchcorn** [comment]
Confirmed duplicate of #7270. Thanks for the heads up @stretchkennedy
